src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs

```diff
@@ -88,27 +150,49 @@ public override void ExecuteCmdlet()
+                GenericResource storageAccountResource = GetStorageAccountResource(StorageAccountName,
+                    StorageAccountResourceGroupName);
                 WriteDebug(string.Format("StorageId = {0}", storageAccountResource.Id));
+                providerParameters.Add(RestoreBackupItemParams.StorageAccountId, storageAccountResource.Id);
+                providerParameters.Add(RestoreBackupItemParams.StorageAccountLocation, storageAccountResource.Location);
+                providerParameters.Add(RestoreBackupItemParams.StorageAccountType, storageAccountResource.Type);
+
+                if (ParameterSetName == AzureFileParameterSet && TargetStorageAccountName != null)
+                {
+                    GenericResource targetStorageAccountResource = GetStorageAccountResource(TargetStorageAccountName,
+                        TargetStorageAccountResourceGroupName);
+                    WriteDebug(string.Format("StorageId = {0}", targetStorageAccountResource.Id));
+                    providerParameters.Add(RestoreFSBackupItemParams.TargetStorageAccountId,
+                        targetStorageAccountResource.Id);
+                    providerParameters.Add(RestoreFSBackupItemParams.TargetStorageAccountLocation,
+                        targetStorageAccountResource.Location);
+                }
 
-                Dictionary<Enum, object> providerParameters = new Dictionary<Enum, object>();
                 providerParameters.Add(VaultParams.VaultName, vaultName);
                 providerParameters.Add(VaultParams.ResourceGroupName, resourceGroupName);
                 providerParameters.Add(VaultParams.VaultLocation, VaultLocation);
                 providerParameters.Add(RestoreBackupItemParams.RecoveryPoint, RecoveryPoint);
-                providerParameters.Add(RestoreBackupItemParams.StorageAccountId, storageAccountResource.Id);
-                providerParameters.Add(RestoreBackupItemParams.StorageAccountLocation, storageAccountResource.Location);
-                providerParameters.Add(RestoreBackupItemParams.StorageAccountType, storageAccountResource.Type);
-                providerParameters.Add(RestoreBackupItemParams.OsaOption, UseOriginalStorageAccount.IsPresent);
+                providerParameters.Add(RestoreVMBackupItemParams.OsaOption, UseOriginalStorageAccount.IsPresent);
+                providerParameters.Add(RestoreFSBackupItemParams.ResolveConflict, ResolveConflict.ToString());
+                providerParameters.Add(RestoreFSBackupItemParams.SourceFilePath, SourceFilePath);
+                providerParameters.Add(RestoreFSBackupItemParams.TargetStorageAccountName, TargetStorageAccountName);
+                providerParameters.Add(RestoreFSBackupItemParams.TargetFileShareName, TargetFileShareName);
+                providerParameters.Add(RestoreFSBackupItemParams.TargetFolder, TargetFolder);
 
                 if (TargetResourceGroupName != null)
```

Comment:
I think we need to check for the ParamSetName instead of individual parameters to populate the right parameters in providerParameters.